scoreposter.py

```diff
@@ -180,38 +157,51 @@ def take_screenshot(url, crop_coordinates):
         raise FileNotFoundError
 
 def getScoreLink(score_id, gamemode):
-    if score_id != None:
-        return f"**Score link:** https://osu.ppy.sh/scores/{mode_to_url_string(int(gamemode))}/{score_id}"
-    else:
+    if score_id == None:
         return None
 
-def scorepost(username : str, ruleset : str):
+    # map the gamemode values to URL strings
+    mode_mapping = {
+        0: "osu",
+        1: "taiko",
+        2: "fruits",
+        3: "mania"
+    }
+    
+    # look up the URL string for the given gamemode
+    mode_string = mode_mapping.get(gamemode, "osu")
```

Comment:
I believe here we should use the .env variable with `string_to_mode()` instead of just using osu! as default